### PR TITLE
[WIP]: ddp pickle fix for learning rate finder

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/1831
This fixes the pickle error on ddp for the learning rate finder as described in the issue. I can get the learning rate finder to finish its search, however since the internal state is destroyed after ddp training so are the logged results. How can I get around this?

### CODE REVIEW

**Comment by williamFalcon** [comment]
i think it's time to add a shared memory queue before going into the process so we can bring stuff back out. this would also solve the test results issue.

Want to add that to this PR?

https://pytorch.org/docs/stable/notes/multiprocessing.html#reuse-buffers-passed-through-a-queue.

https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Queue

**Comment by williamFalcon** [comment]
@SkafteNicki shall we merge this now and tackle the Q thing later?
or shall we push this fix to 0.7.7?

**Comment by SkafteNicki** [comment]
Push to 0.7.7 since the PR right now only is the solution to the pickle problem.
